### DISCUSSION TITLE
change pots

### DISCUSSION THREAD

**Discussion by tiretrak** [marcel-licence/esp32_basic_synth#73] [Q&A]
i use 10k pots.what needs to be changed in the code for the 10k potentiometers to work correctly?

**Comment by marcel-licence** [comment]
Hey it should work without any modification. The analog input measures only the voltage which will be between GND and VCC

**Reply by tiretrak** [reply]
10k works, but the change in values ​​is too fast and too large

**Comment by tiretrak** [comment]
[https://soundcloud.com/tiretrak/10k-pots-1?si=c8c1d2751f3d47938cb2195fc2aa1fcb](url)

**Reply by marcel-licence** [reply]
Oh I hear the problem (the link didn't work but I could find it on your soundcloud page). Do you use linear or logarithmic pots?
Do you use a multiplexer to use more than one pot?
Could you measure the voltage at the ADC input?

By the way - sounds really good! Did you use external effects?

**Reply by tiretrak** [reply]
i use RK09D1130C2P pots. linear.
i`m using 4067 multiplexer.
ok.i try this.
yes, it`s external effects - adau 1467 dsp, connection by i2s.

**Reply by tiretrak** [reply]
>Could you measure the voltage at the ADC input?
0-3.33v on middle pin on pot
multiplexer board vcc - 3.33v

**Reply by marcel-licence** [reply]
Sounds correct. I wanted to figure out if the input voltage behaves like what you are hearing or the calculation or maybe the multiplexing fails. The safest method would be to test just one pot directly connected to the ADC without multiplexing. 
By adding
`#define ADC_DEBUG_CHANNEL0_DATA`
you should get a debug output by moving the pot. 

Your observed effect can also be caused by floating lines. I also observed during testing that neighbor channels influence each other in case nothing is connected. Do you have any pins left open?

**Comment by tiretrak** [comment]
>Do you have any pins left open?
yes

**Reply by marcel-licence** [reply]
yes input 9-16 are unused. All other pins are connected (8 pots, EN->GND, S1-S3 to the Audio Kit and S4->GND or VCC). If S4 or EN is floating you can get strange effects also influenced by the pot rotation.

**Comment by tiretrak** [comment]
S3 ang EN (adc board) connect to GND .
what is S4 ?

**Reply by marcel-licence** [reply]
Okay that is correct. I meant S3 instead of S4. And S0-S2 connected to the audio kit.
Did you try out to enable the debug output? In addition to that connect a pot directly to IO12?

**Comment by tiretrak** [comment]
c9-c15 on adc board is floating
